spamFilter.go

```diff
@@ -145,24 +179,42 @@ func isSpam(mailString string, probMap map[string]float64) bool {
 	}
 	//quickSort(wordProbSlice, 0, len(wordProbSlice) - 1)
 	sort.Sort(ByInterest(wordProbSlice))
-	mostInteresting := wordProbSlice[len(wordProbSlice) - 15:]
-	var probProd, invProbProd float64 = 1.0
+	//fmt.Println("the least to most interesting words are: ", wordProbSlice)
+	var mostInteresting []wordProb
+	if len(wordProbSlice) > 15 {
+		mostInteresting = wordProbSlice[len(wordProbSlice) - 15:]
+	} else {
+		mostInteresting = wordProbSlice 
+	}
+	//fmt.Println("the most interesting words are: ", mostInteresting)
+	var probProd, invProbProd float64 = 1.0, 1.0
 	for _, v := range mostInteresting {
-		probProd = v.probability * probProd
+		probProd *= v.probability
 		invProb := 1 - v.probability
-		invProbProd = invProb * invProbProd
+		invProbProd *= invProb
 	}	
-	combProd := probProd/(probProd + invProbProd)
+	combProb := probProd/(probProd + invProbProd)
+	//fmt.Println("the combined probability is: ", combProb)
 	var mailIsSpam bool
-	if combProd >= 0.9 {
+	if combProb >= 0.9 {
 		mailIsSpam = true
 	} else {
 		mailIsSpam = false
 	}
 	return mailIsSpam
 }
 
+
 func main() {
-	//fileString := fileToString()
-	//fmt.Println(wordFreq(fileString))
+	goodCorpus, nGoodMail := buildCorpus("/Users/moose1/Documents/SpamFilter/lingspam_public/NotSpam")
```

Comment:
If you're interested, you could extend this program to take notspam and spam corpuses as command line flags as well.